### ISSUE TITLE
SharedHttpCacheStorage doesnt resolve redirect correctly if the uri that is given isn't normalized

### ISSUE THREAD

**Issue by jcompagner** [eclipse-tycho/tycho#2666]
The getCacheEntry function:
https://github.com/eclipse-tycho/tycho/blob/master/p2-maven-plugin/src/main/java/org/eclipse/tycho/p2maven/transport/SharedHttpCacheStorage.java#L92

should normalize the incoming url, so that will not become an extra redirect that the HttpClient of java does (and then 2 redirects must be followed what the code of Tycho doesn't do and then you get a stack overflow because getCachedEntry is constantly called by itself through the redirect code.

```
	public CacheEntry getCacheEntry(URI uriParam, Logger logger) throws FileNotFoundException {
		URI uri = uriParam.normalize();
```

that is the code that fixes it, the param is renamed and the an variable uri is made that is the normalized version of it.
Then the Java HttpClient doesn't treat this as a extra redirect.

**Comment by laeubi** [comment]
@jcompagner can this be closed now or do you still see issues?

**Comment by jcompagner** [comment]
yes this one is fixed 4.0.1 works fine now for our product build
